solr/core/src/java/org/apache/solr/schema/IndexSchema.java

```diff
@@ -1991,6 +1997,37 @@ public IndexSchema deleteCopyFields(Map<String, Collection<String>> copyFields)
     throw new SolrException(ErrorCode.SERVER_ERROR, msg);
   }
 
+  /** Informs analyzers used by a fieldType. */
+  protected void informResourceLoaderAwareObjectsForFieldType(FieldType fieldType) {
+    // must inform any sub-components used in the
+    // tokenizer chain if they are ResourceLoaderAware
+    if (!fieldType.supportsAnalyzers()) return;
+
+    Analyzer indexAnalyzer = fieldType.getIndexAnalyzer();
```

Comment:
maybe do something like:

```java
Stream.of(fieldType.getIndexAnalyzer(), fieldType.getQueryAnalyzer(), noIdeaHowToHandleThis.getMultiTermAnalyzer()).filter(a -> a instanceof TokenizerChain)
  .distinct().foreach(a -> informResourceLoaderAwareObjectsInChain((TokenizerChain) a);
```

Reply:
maybe this one works:

```java
var stream = Stream.of(fieldType.getIndexAnalyzer(), fieldType.getQueryAnalyzer());
if (fieldType instanceof TextField) {
  stream = Stream.concat(stream, Stream.of(((TextField) fieldType).getMultiTermAnalyzer());
}
stream.filter(a -> a instanceof TokenizerChain)
  .distinct().foreach(a -> informResourceLoaderAwareObjectsInChain((TokenizerChain) a);
```

Reply:
(the distinct is by equals, but by default they use system's equals)

Reply:
null check is not needed because of `TokenizerChain.class::isInstance` (returns false on `null`)

Reply:
Note that this entire method is *moved* over from `ManagedIndexSchema`, and I have not modified it or attempted to optimize it. May we leave improvements to anoter PR?

Reply:
Ah, you're right. I just thought al little bit of deduplication of code would be fine.

It's up to you to apply that, but I can make a separate PR.

Reply:
I'll close this comment and defer to new PR after this landed